core/client/app/components/gh-menu-toggle.js

```diff
@@ -0,0 +1,47 @@
+/*
+This cute little component has two jobs.
+
+On desktop, it toggles autoNav behaviour. It tracks
+that state via the maximise property, and uses the
+state to render the appropriate icon.
+
+On mobile, it renders a closing icon, and clicking it
+closes the mobile menu
+*/
+
+import Ember from 'ember';
+import mobileQuery from 'ghost/utils/mobile';
+
+export default Ember.Component.extend({
+    classNames: ['gh-menu-toggle'],
+
+    isMobile: false,
+    maximise: false,
+
+    iconClass: Ember.computed('maximise', 'isMobile', function () {
+        if (this.get('maximise') && !this.get('isMobile')) {
+            return 'icon-maximise';
+        } else {
+            return 'icon-minimise';
+        }
+    }),
+
+    init () {
+        var self = this;
+        this._super(...arguments);
+        this.set('isMobile', mobileQuery.matches);
+        mobileQuery.addListener(() =>
+            Ember.run.next(() => self.set('isMobile', mobileQuery.matches))
+        );
+    },
+
+    click: function () {
+        if (this.get('isMobile')) {
+            this.sendAction('mobileAction');
+        }
+        else {
+            this.toggleProperty('maximise');
```

Comment:
Really, I'm not sure that the component should be tracking this state since it belongs to the application controller. But I also don't want to be passing controller state three components down ;) Anyone else have an opinion?